pkg/agent/controller/egress/egress_controller.go

```diff
@@ -1118,39 +1118,42 @@ func (c *EgressController) syncEgress(egressName string) error {
 	}()
 
 	egressIP := net.ParseIP(eState.egressIP)
-	// Install SNAT flows for desired Pods.
-	for pod := range pods {
-		eState.pods.Insert(pod)
-		stalePods.Delete(pod)
+	egress, _ = c.egressLister.Get(egressName)
+	if egress.Status.EgressNode != "" {
```

Comment:
As I pointed out before, I do feel like it is strange to consume `.Status.EgressNode` as part of the controller, and I am not sure this feature is very useful, but I will let @tnqn review & decide.

Reply:
I'm not sure if this is an improvement either. Note that if this behavior change isn't wanted by all users, it would satisfy some while breaking others. The issue being addressed doesn't come from end users so I would prefer keeping it as was.
Instead of changing the behavior from one to another, I would prefer making it explicit and configurable, letting user know and decide what would happen.

Reply:
okay so in that case we can have a config variable set by user in antrea.yml and if that i set then we can allow normal Node SNAT if static egress is not assigned to any node, and if the variable is not set or is false then we will keep the behaviour as it is.
Is it what is expected @antoninbas @tnqn

Reply:
We can add a config, but I think the other parts of https://github.com/antrea-io/antrea/issues/6228 should also be addressed by this PR:
1) consistent behavior for static Egress and HA Egress
2) config parameter also applies to HA Egress

And I think there should be e2e testing for HA Egress at least

Reply:
@antoninbas @jainpulkit22 would it make sense to add a field in the egress object? Because applications may have different expectations—some may prefer EgressIP and can accept NodeIP as fallback, while others may require strict enforcement.

Reply:
I don't feel strongly either way. Maybe we should have the ability for the cluster admin to set a default in the Antrea config, and the ability to change that behavior on a per-Egress basis using a field (which is similar to how we let users change the LoadBalancer mode for Services). Then cluster admins want to prevent application owners from deviating from the default behavior when they create Egress resources, they can deploy an OPA policy for Egress.

Reply:
So, as a conclusion we need to make both Egress configurable based on the config variable set in the Antrea config?